zeebe/protocol/src/main/java/io/camunda/zeebe/protocol/record/intent/UserTaskIntent.java

```diff
@@ -40,7 +40,9 @@ public enum UserTaskIntent implements ProcessInstanceRelatedIntent {
   UPDATING(12),
   UPDATED(13),
 
-  MIGRATED(14);
+  MIGRATED(14),
+
+  COMPLETE_TASK_LISTENER(15);
```

Comment:
❓ Did we consider alternative names? For example, `CONTINUE_AFTER_TASK_LISTENER_COMPLETED`

Reply:
I followed the same naming convention that we use for execution listeners,
specifically [`ProcessInstanceIntent.COMPLETE_EXECUTION_LISTENER`](https://github.com/camunda/camunda/blob/7add7a0dc05d08d11b07f8b6b1764501b9a9c169/zeebe/protocol/src/main/java/io/camunda/zeebe/protocol/record/intent/ProcessInstanceIntent.java#L39), which was introduced to serve the same purpose —> freezing the process execution until this command is written.

That’s why I opted for `COMPLETE_TASK_LISTENER` to maintain consistency.

However, I agree that your option, `CONTINUE_AFTER_TASK_LISTENER_COMPLETED`, sounds really self-explanatory and may actually suit this case better! 👍

I’m open to discussing if we decide to go with a more descriptive naming!

Reply:
💭 Although possible, it's a bit hard to change this. You'd need to add the new intent name, mark the old as deprecated, and rename the processor and register it for both the deprecated and the new intent (for backward compatibility). I'm not sure whether it's worth it. I'll leave that up to you. As an alternative, we could consider documenting the meaning in JavaDoc on the intent.